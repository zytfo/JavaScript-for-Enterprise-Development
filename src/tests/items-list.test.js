import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import createStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import ItemsList from '../components/items-list'

describe('ItemsList', () => {
    it('test render empty component', () => {
        const mockStore = createStore([thunk]);
        const initialState = {
            data: []
        };
        const store = mockStore(initialState);
        const container = renderer
            .create(
                <Provider store={store}>
                    <Router>
                        <Route component={ItemsList}/>
                    </Router>
                </Provider>,
                { createNodeMock: ({ type }) => document.createElement(type) }
            );
        expect(container.toJSON()).toMatchSnapshot()
    })
});

