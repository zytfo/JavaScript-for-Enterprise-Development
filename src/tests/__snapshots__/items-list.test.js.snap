// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList test render empty component 1`] = `
<div
  className="body"
>
  <div
    className="container row"
  />
</div>
`;
